src/topology_sampler.hpp

```diff
@@ -0,0 +1,129 @@
+// Copyright 2019-2021 bito project contributors.
+// bito is free software under the GPLv3; see LICENSE file for details.
+
+#pragma once
+
+#include "node.hpp"
+#include "mersenne_twister.hpp"
+#include "eigen_sugar.hpp"
+
+class TopologySampler {
+public:
+
+  class Input {
```

Comment:
This is very cool! @ognian- -- is there a name for this pattern?

Reply:
I'm not aware of a pattern name for this particular setup, it's simply an interface class for the algorithm input. I chose run-time polymorphism instead of template because the overhead is minimal, but it will reduce compilation time and header dependencies.